src/middleware/authorisation.coffee

```diff
@@ -88,56 +88,72 @@ if process.env.NODE_ENV == "test"
 exports.isChannelEnabled = isChannelEnabled = (channel) -> not channel.status or channel.status is 'enabled'
 
 
+channelWithHighestPriority = (channels) ->
```

Comment:
This needs to be unit tested thoroughly so that we can verify all it's behaviours.